crates/shared/src/trade_finding/external.rs

```diff
@@ -112,7 +112,7 @@ impl From<dto::Quote> for Trade {
 
         Self {
             out_amount: quote.amount,
-            gas_estimate: TRADE_GAS,
+            gas_estimate: quote.gas.unwrap_or(TRADE_GAS),
```

Comment:
Don't we need to be more strict here and discard solutions that we cannot verify and that don't have a gas estimate?